spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/partition/DeployerPartitionHandlerTests.java

```diff
@@ -79,14 +80,17 @@
 	private StepExecutionSplitter splitter;
 
 	@Mock
-	private JobRepository jobRepository;
+	private TaskRepository taskRepository;
 
 	private Environment environment;
 
 	@Before
 	public void setUp() {
 		MockitoAnnotations.initMocks(this);
 		this.environment = new MockEnvironment();
+		TaskExecution taskExecution = new TaskExecution(2, 0, "name", new Date(),
+				new Date(), "", Collections.emptyList(), null, null, null);
+		when(taskRepository.createTaskExecution()).thenReturn(taskExecution);
 	}
 
 	@Test
```

Comment:
We are only doing a test on the old constructor's validation.  We need one for the new constructor as well, especially since the new one does not use the old one (although it could).

Reply:
The new test added only tests it's validation...not that it still works with the new code (see my comments above).